app-agent/src/exec_process.rs

```diff
@@ -33,6 +54,198 @@ pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result
+    } else if user_perm == "r" || group_perm == "r" || other_perm == "r" {
+        log::error!("file '{}' is missing the following permissions:  'r'", external_command);
+        log::info!("💡 Hint: try 'chmod +r {}'", external_command)
+    } else if user_perm == "x" || group_perm == "x" || other_perm == "x" {
+        log::error!("file '{}' is missing the following permissions:  'x'", external_command);
+        log::info!("💡 Hint: try 'chmod +x {}'", external_command)
+    } else {
+        log::warn!("Can't determine right issue about the file: {}", external_command);
+    }
+    "Issue happened about file's permission".to_string()
+}
+
+fn handle_not_found(external_command: String, args: Vec<String>) -> String {
+    fn resolve_application_path() -> std::io::Result<PathBuf> {
+        std::env::current_exe()?.canonicalize()
+    }
+    log::error!("Command '{}' not found", external_command);
+    let directory_entries_iter = match fs::read_dir(".") {
+        Ok(directory) => directory,
+        Err(err) => {
+            panic!("Error when trying to read current directory: {}", err);
+        }
+    };
+    let app_path = resolve_application_path()
+        .ok()
+        .and_then(|p| p.to_str().map(|s| s.to_owned()))
+        .unwrap_or(String::from("path/to/agent"));
+
+    let mut lowest_distance = usize::MAX;
+    let mut best_element = None;
+
+    for entry_result in directory_entries_iter {
+        let entry = entry_result.unwrap();
+        let entry_type = entry.file_type().unwrap();
+        if entry_type.is_file() {
+            let entry_string = entry.file_name().into_string().unwrap();
+            let distance = super::utils::distance_with_adjacent_transposition(
+                external_command
+                    .strip_prefix("./")
+                    .unwrap_or(&external_command)
+                    .to_string(),
+                entry_string.clone(),
+            );
+            if distance < 3 && distance < lowest_distance {
+                lowest_distance = distance;
+                best_element = Some((entry_string, distance));
+            }
+        }
+    }
+    match best_element {
+        Some((element, distance)) => {
+            if distance == 0 {
+                log::info!(
+                    "💡 Hint: A file named '{}' exists in the current directory. Prepend ./ to execute it.",
+                    element
+                );
+                log::info!(
+                    "Example: {} exec ./{} {}",
+                    app_path,
+                    element,
+                    args.iter()
+                        .map(|arg| {
+                            if arg.contains(' ') {
+                                format!("\"{}\"", arg)
+                            } else {
+                                arg.to_string()
+                            }
+                        })
+                        .collect::<Vec<_>>()
+                        .join(" ")
```

Comment:
Please separate the construction of the list from the log call (same everywhere).